addon/models/osf-model.js

```diff
@@ -56,7 +56,7 @@ export default DS.Model.extend({
         this.eachRelationship((rel) => {
             var relation = this.resolveRelationship(rel);
             // TODO(samchrisinger): not sure if hasLoaded is a subset if the hasData state
-            if (relation.hasData && relation.hasLoaded) {
+            if (relation.hasData && (relation.hasLoaded || rel === 'affiliatedInstitutions')) {
```

Comment:
@jamescdavis, we talked about this. Do you happen to remember why this was the case?

Reply:
@aaxelb ok. Totally remember why. So, hasLoaded is a flag used to see if we have loaded models for the relationship (https://github.com/emberjs/data/blob/b329a2c7018b86716d202610d3b6d0e080e370da/addon/-private/system/relationships/state/relationship.js#L475). However, institution relationships use a piece of json-api (http://jsonapi.org/format/#crud-updating-to-many-relationships) that dictates a shorthand for one to many relationships, that allow us to get and update a model's relationship by retrieving data like this:
```

{
  "data": [
    { "type": "tags", "id": "2" },
    { "type": "tags", "id": "3" }
  ]
}
```
This is all simple and fine when talking about how one model relates to another, but ember-data expects an all-attributes account of a model when it is retrievied via a relationship. So, updating the relationship works, but `affiliatedInstitutions` are never `hasLoaded` because it did not load all the information required for the institution model. However, for those kinds of relationships, as mentioned above, we actually dont need the entire model to make a request. So, the exception. Does this seem right, @aaxelb ?

Reply:
Example in the wild (https://api.osf.io/v2/nodes/myanx/relationships/institutions/)

Reply:
Hmm, I guess that makes sense. Just wish there were a more general way to detect it than special-casing the name of the relationship. Is this the only place where the API uses that sort of shorthand?

(TIL https://api.osf.io/v2/nodes/myanx/relationships/institutions/ and https://api.osf.io/v2/nodes/myanx/institutions/ both exist and do almost, but not quite, entirely different things)

Reply:
Good question. Looking into it now

Reply:
It seems we implement the same thing for linked_nodes and linked_registrations, although would be good to do more looking into to find out whether we use the `/relationships/linked_nodes` or the `/linked_nodes/` endpoints, and why, and how hard (possible??) it would be to get institutions to use `/institutions`. Can look into it tomorrow